src/App.tsx

```diff
@@ -15,37 +17,122 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  NONE,
+  ALPHABET,
+  LENGTH,
+}
+
 export const App: React.FC = () => {
+  const [goods, setGoods] = useState(goodsFromServer);
+  const [sortType, setSortType] = useState(SortType.NONE);
+  const [reverse, setReverse] = useState(false);
+  const [clicked1, setClicked1] = useState(false);
+  const [clicked2, setClicked2] = useState(false);
+  const [clicked3, setClicked3] = useState(false);
+  const [clicked4, setClicked4] = useState(false);
+
+  useEffect(() => {
+    const sortedGoods = [...goodsFromServer];
+
+    if (sortType === SortType.ALPHABET) {
+      sortedGoods.sort((a, b) => a.localeCompare(b));
+    } else if (sortType === SortType.LENGTH) {
+      sortedGoods.sort((a, b) => a.length - b.length);
+    }
+
+    if (reverse) {
+      sortedGoods.reverse();
+    }
+
+    setGoods(sortedGoods);
+  }, [sortType, reverse]);
+
+  const resetClicked = () => {
+    setClicked1(false);
+    setClicked2(false);
+    setClicked3(false);
+    setClicked4(false);
+  };
+
+  const handleSort = (type: SortType) => {
+    setSortType(type);
+    setReverse(false);
+    resetClicked();
+  };
+
+  const handleReverse = () => {
+    setReverse(!reverse);
```

Comment:
```suggestion
    setReverse(curr => !curr);
```
Always use callback to set state based on the previous one